src/expr/term_conversion_proof_generator.cpp

```diff
@@ -40,36 +40,14 @@ TConvProofGenerator::~TConvProofGenerator() {}
 
 void TConvProofGenerator::addRewriteStep(Node t, Node s, ProofGenerator* pg)
 {
-  if (t == s)
-  {
-    // nothing to do
-    return;
-  }
-  // should not rewrite term to two different things
-  if (hasRewriteStep(t))
-  {
-    Assert(getRewriteStep(t) == s);
-    return;
-  }
-  Node eq = t.eqNode(s);
+  Node eq = registerRewriteStep(t,s);
```

Comment:
If you have `registerRewriteStep(t,s);` return a null node and not check that will screw up the lines below, no?

Reply:
ah yeah, whoops.